src/Nethermind/Nethermind.TxPool/TransactionExtensions.cs

```diff
@@ -74,7 +78,18 @@ internal static bool IsOverflowWhenAddingTxCostToCumulative(this Transaction tx,
             overflow |= UInt256.AddOverflow(currentCost, maxTxCost, out cumulativeCost);
             overflow |= UInt256.AddOverflow(cumulativeCost, tx.Value, out cumulativeCost);
 
+            if (tx.SupportsBlobs)
+            {
+                // if tx.SupportsBlobs and has BlobVersionedHashes = null, it will throw on earlier step of validation, in TxValidator
+                overflow |= UInt256.MultiplyOverflow(Eip4844Constants.BlobGasPerBlob, (UInt256)tx.BlobVersionedHashes!.Length, out UInt256 blobGas);
+                overflow |= UInt256.MultiplyOverflow(blobGas, tx.MaxFeePerBlobGas ?? UInt256.MaxValue, out UInt256 blobGasCost);
```

Comment:
Two lines above may become a part of BlobGasCalculator just to have one place for such things I guess